helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -333,15 +338,19 @@ private void forceRebalance(HelixManager manager, ClusterEventType eventType) {
   void startPeriodRebalance(long period, HelixManager manager) {
     if (period != _timerPeriod) {
```

Comment:
I think this check needs to be included in the critical section too.

Reply:
I think including this check in won't have any functional different compared to not including it. For case like 2 thread change value from v0->v1 and v0->v2 at the same time, the result is indeterministic in either way.  I prefer critical section like this since it excludes 2 log IOs.

Reply:
Second JJ's comment. IMO since the set of this value is inside critical section, the read should be included too. Theoretically you could get a bad value.

Reply:
TFTR. I think you are referring to situations like two threads both try to change period to a same value. including this compare in the inside critical section or not won't have different result. 
For the case when we change period to different values, the result is indeterminable in either way. 
Please correct m if I am wrong. I think in conclusion, having the compare in or out of critical section will have the same behavior. So I prefer smallest  critical section in this case, especially we have two IOs here.